lib/matplotlib/__init__.py

```diff
@@ -605,7 +605,7 @@ def gen_candidates():
 )
 class RcParams(MutableMapping, dict):
```

Comment:
I may not fully get the scope here, but I take it that one reason to do this is that we should not subclass from `dict` later on? Or is the whole idea to provide a stable API for this, but actually still subclass?

Reply:
We want to get rid of dict subclassing eventually. Using a dict subclass dictates the internal data model. We likely don't want this in the future when we're remodelling the config data structure. Either the new structure will be a 100% API compatible drop in and the config object will be available via `rcParams`; or we have a new completely free to design `config` object and `rcParams` will loose all state and only be an adapter to that new `config` object. Either way, being bound to a dict subclass would be cumbersome.